backend/dataall/modules/maintenance/services/maintenance_service.py

```diff
@@ -0,0 +1,149 @@
+"""
+A service layer for maintenance activity
+Defines functions and business logic to be performed for maintenance window
+"""
+
+import logging
+import os
+
+from dataall.base.aws.event_bridge import EventBridge
+from dataall.base.aws.parameter_store import ParameterStoreManager
+from dataall.base.context import get_context
+from dataall.core.permissions.services.tenant_policy_service import TenantPolicyValidationService
+from dataall.modules.maintenance.api.enums import MaintenanceStatus
+from dataall.modules.maintenance.db.maintenance_repository import MaintenanceRepository
+from dataall.core.stacks.aws.ecs import Ecs
+
+logger = logging.getLogger(__name__)
+
+
+class MaintenanceService:
+    @staticmethod
+    def start_maintenance_window(mode: str = None):
+        """
+        Start maintenance window by performing following actions
+            1. Perform validation to check if the user belongs to the DAAdministrators group
+            2. Put the maintenance window status to PENDING and update the maintenance mode
+            3. Get all the ECS Scheduled tasks and disable the schedule for them
+        @param mode: mode to set for maintenance window
+        @return: returns True if successful or False
+        """
+        # Check from the context if the groups contains the DAAAdminstrators group
+        groups = get_context().groups if get_context().groups is not None else []
```

Comment:
I thought get_context().groups returned an empty list when the user belonged to no group.... good finding

Reply:
`groups: list = extract_groups(user_id, claims)`?

Reply:
This is been used only for initializing the groups list when the API handler is called. This takes in the claims which come from the cognito/custom authorizer.